integration/openshift/e2e/gen_test_cases.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# This script should be used to generate the list of test cases from
+# the OpenShift e2e conformance suite.
+#
+# Edit the filter_patterns, exclude_patterns, and skip_patterns variables
+# and re-run this script in order to change the tests selection.
+#
+set -e
+set -o nounset
+
+GOPATH=${GOPATH:-$(go env GOPATH)}
+script_dir="$(realpath $(dirname $0))"
+tests_dir="$script_dir/../../../"
+source "$tests_dir/lib/common.bash"
+source "$script_dir/../lib.sh"
+
+# Tests that should be included. Leave it empty to pick them all.
+# Use grep-like extended regular expressions in quotes.
+filter_patterns=(
+        )
+
+# Tests that should be excluded.
+exclude_patterns=(
+        )
+
+# Tests that would normally be executed but due to some issue they should not.
+skip_patterns=(
+        )
+
+# Define temporary files and write down the patterns. Don't delete the files
+# afterwards because they can be used to debug.
+#
+inclusion_file="${script_dir}/tests_inclusion_patterns"
+exclusion_file="${script_dir}/tests_exclusion_patterns"
+skip_file="${script_dir}/tests_skip_patterns"
+tmp_tests_file="${script_dir}/new_test_cases"
+test_cases_file="${script_dir}/test_cases.txt"
+
+# The OpenShift e2e suite used to generate our list of tests.
+suite="openshift/conformance/parallel"
+
+function setup()
+{
+	# Build the binary if not found on PATH.
+	#
+	openshift_tests=$(build_openshift_tests) || \
+		        die "Failed to build openshift-tests"
+
+	printf "%s\n" "${filter_patterns[@]}" > "${inclusion_file}"
+	printf "%s\n" "${exclude_patterns[@]}" > "${exclusion_file}"
+	printf "%s\n" "${skip_patterns[@]}" > "${skip_file}"
+}
+
+function cleanup()
+{
+	rm -f "${inclusion_file}" "${exclusion_file}" "${skip_file}"
+}
+
+trap cleanup EXIT
+
+function print_header()
+{
+	cat <<EOF
```

Comment:
Nit: You could use `<<-EOF` and then tab indent the here doc lines to make the function look neater ;)